x/twap/logic_test.go

```diff
@@ -539,316 +537,6 @@ func (s *TestSuite) TestGetInterpolatedRecord_ThreeAsset() {
 	}
 }
 
-type computeTwapTestCase struct {
-	startRecord types.TwapRecord
-	endRecord   types.TwapRecord
-	twapTypes   []twap.TwapType
-	quoteAsset  string
-	expTwap     sdk.Dec
-	expErr      bool
-	expPanic    bool
-}
-
-type computeThreeAssetArithmeticTwapTestCase struct {
-	startRecord []types.TwapRecord
-	endRecord   []types.TwapRecord
-	quoteAsset  []string
-	expTwap     []sdk.Dec
-	expErr      bool
-}
-
-// TestComputeArithmeticTwap tests computeTwap on various inputs.
-// TODO: test both arithmetic and geometric twap.
-// The test vectors are structured by setting up different start and records,
-// based on time interval, and their accumulator values.
-// Then an expected TWAP is provided in each test case, to compare against computed.
-func TestComputeTwap(t *testing.T) {
```

Comment:
We should keep this test case here as well while refactoring it to take in strategy as an argument